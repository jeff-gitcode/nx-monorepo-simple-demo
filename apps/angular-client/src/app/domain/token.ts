export class Token {
  accessToken?: string;
  refreshToken?: string;

  constructor() {
    this.accessToken = '';
    this.refreshToken = '';
  }
}
