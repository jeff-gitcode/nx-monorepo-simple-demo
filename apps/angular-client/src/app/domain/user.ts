export class UserDTO {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  accessToken?: string;
  refreshToken?: string;

  constructor() {
    this.id = '';
    this.firstName = '';
    this.lastName = '';
    this.email = '';
    this.password = '';
    this.accessToken = '';
    this.refreshToken = '';
  }
}

export class UserInput {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  accessToken?: string;
  refreshToken?: string;

  constructor() {
    this.id = '';
    this.firstName = '';
    this.lastName = '';
    this.email = '';
    this.password = '';
    this.accessToken = '';
    this.refreshToken = '';
  }
}

export class LoginUser {
  username!: string;
  password!: string;

  constructor() {
    this.username = '';
    this.password = '';
  }
}

export class LoginUserDTO {
  email!: string;
  password!: string;

  constructor() {
    this.email = '';
    this.password = '';
  }
}
