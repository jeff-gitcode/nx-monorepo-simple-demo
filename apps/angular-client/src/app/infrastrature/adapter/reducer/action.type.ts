export enum ActionTypes {
  // user list actions
  getUserList = 'GET_USER_LIST',
  getUserListSuccess = 'GET_USER_LIST_SUCCESS',
  getUserListFailure = 'GET_USER_LIST_FAILURE',

  // user CRUD actions
  getUser = 'GET_USER',
  getUserSuccess = 'GET_USER_SUCCESS',
  getUserFailure = 'GET_USER_FAILURE',

  createUser = 'CREATE_USER',
  createUserSuccess = 'CREATE_USER_SUCCESS',
  createUserFailure = 'CREATE_USER_FAILURE',

  updateUser = 'UPDATE_USER',
  updateUserSuccess = 'UPDATE_USER_SUCCESS',
  updateUserFailure = 'UPDATE_USER_FAILURE',

  deleteUser = 'DELETE_USER',
  deleteUserSuccess = 'DELETE_USER_SUCCESS',
  deleteUserFailure = 'DELETE_USER_FAILURE',

  // json form actions
  getJsonForm = 'GET_JSON_FORM',
  getJsonFormSuccess = 'GET_JSON_FORM_SUCCESS',
  getJsonFormFailure = 'GET_JSON_FORM_FAILURE',

  // auth actions
  login = 'LOGIN',
  loginSuccess = 'LOGIN_SUCCESS',
  loginFailure = 'LOGIN_FAILURE',

  signup = 'SIGN_UP',
  signupSuccess = 'SIGN_UP_SUCCESS',
  signupFailure = 'SIGN_UP_FAILURE',

  refresh = 'REFRESH',
  refreshSuccess = 'REFRESH_SUCCESS',
  refreshFailure = 'REFRESH_FAILURE',

  logout = 'LOGOUT',
  logoutSuccess = 'LOGOUT_SUCCESS',
  logoutFailure = 'LOGOUT_FAILURE',
}
