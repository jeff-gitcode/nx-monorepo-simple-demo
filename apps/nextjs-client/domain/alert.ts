import { AlertType } from '../infrastructure/service/alert.service';

export interface AlertMessage {
  id?: string;
  type: AlertType;
  message: string;
}
