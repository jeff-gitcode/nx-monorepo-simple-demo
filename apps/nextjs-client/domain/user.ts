export interface UserDTO {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  isDeleting?: boolean;
}

export interface LoginUser {
  username: string;
  password: string;
}
