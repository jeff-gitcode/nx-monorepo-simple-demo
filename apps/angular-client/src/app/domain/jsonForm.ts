import { UserDTO } from './user';

export type FormList = {
  accessor: keyof UserDTO;
  Header: string;
};

export type Controls = {
  name: string;
  label: string;
  value: string;
  type: string;
  validators: any;
};

export class JsonForm {
  formFields!: Controls[];
  formList!: FormList[];
}
