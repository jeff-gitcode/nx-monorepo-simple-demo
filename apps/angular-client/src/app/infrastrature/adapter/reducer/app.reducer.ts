import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { ActionReducer, ActionReducerMap } from '@ngrx/store';
import { storeLogger } from 'ngrx-store-logger';
import { authReducer, IAuthState, initialAuthState } from './auth.reducer';
import {
  IJsonFormState,
  initialJsonFormState,
  jsonFormReducer,
} from './jsonform.reducer';

import { initialUserState, IUserState, userReducer } from './user.reducer';

export interface IAppState {
  router?: RouterReducerState;
  users: IUserState;
  jsonForm: IJsonFormState;
  auth: IAuthState;
}

export const initialAppState: IAppState = {
  users: initialUserState,
  jsonForm: initialJsonFormState,
  auth: initialAuthState,
};

export function getInitialState(): IAppState {
  return initialAppState;
}

export const appReducers: ActionReducerMap<IAppState, any> = {
  router: routerReducer,
  users: userReducer,
  jsonForm: jsonFormReducer,
  auth: authReducer,
};

export function logger(reducer: ActionReducer<IAppState>): any {
  // default, no options
  return storeLogger()(reducer);
}

export const metaReducers = [logger]; // environment.production ? [] : [logger];
