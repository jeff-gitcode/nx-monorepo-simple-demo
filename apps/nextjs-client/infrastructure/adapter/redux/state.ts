import { Subscription } from 'react-hook-form/dist/utils/createSubject';
import { AlertMessage } from '../../../domain/alert';

export type FormState = {
  items?: any;
  errors?: string;
};

export type UserListState = {
  items?: any[];
  errors?: string;
};

export type UserState = {
  id?: string;
  items?: any;
  errors?: string;
};

export type AuthState = {
  item?: any;
  loginUser?: any;
  isLogin?: boolean;
  errors?: string;
};

export type AlertState = {
  observer?: any;
  items?: Subscription;
  item?: AlertMessage;
  errors?: string;
};

export interface StateType {
  readonly form: FormState;
  readonly userList: UserListState;
  readonly user: UserState;
  readonly alert: AlertState;
  readonly auth: AuthState;
}

export const initialState: StateType = {
  form: {
    items: {},
    errors: '',
  },
  userList: {
    items: [],
    errors: '',
  },
  user: {
    id: '',
    items: {},
    errors: '',
  },
  alert: {
    observer: {
      unsubscribe: () => {},
    },
    errors: '',
  },
  auth: {
    item: null,
    loginUser: null,
    isLogin: false,
    errors: '',
  },
};
