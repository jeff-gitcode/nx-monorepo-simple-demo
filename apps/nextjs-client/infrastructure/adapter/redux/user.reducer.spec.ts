import { UserDTO } from '../../../domain/user';
import { ActionsTypes } from './action.type';
import { initialState } from './state';
import reducer, {
  createUserAction,
  deleteUserAction,
  updateUserAction,
} from './user.reducer';

describe('reducers/user', () => {
  const emptyAction = { type: '', payload: {} };
  let currentState: any = initialState.user;
  const mockUser: UserDTO = {
    id: '1',
    firstName: 'firstname',
    lastName: 'lastName',
    email: 'email@example.com',
    password: 'password',
  };
  beforeAll(() => {
    currentState = reducer(undefined, emptyAction);
  });

  afterAll(() => {
    currentState = initialState.user;
  });

  it('should return the initial state', () => {
    expect(reducer(currentState, emptyAction)).toMatchSnapshot();
  });

  it(`should handle ${ActionsTypes.createUser}`, () => {
    currentState = reducer(currentState, createUserAction(mockUser));
    expect(currentState).toMatchSnapshot();
  });

  it(`should handle ${ActionsTypes.updateUser}`, () => {
    currentState = reducer(currentState, updateUserAction(mockUser));
    expect(currentState).toMatchSnapshot();
  });

  it(`should handle ${ActionsTypes.deleteUser}`, () => {
    currentState = reducer(currentState, updateUserAction(mockUser));
    currentState = reducer(currentState, deleteUserAction(1));
    expect(currentState).toMatchSnapshot();
  });
});
