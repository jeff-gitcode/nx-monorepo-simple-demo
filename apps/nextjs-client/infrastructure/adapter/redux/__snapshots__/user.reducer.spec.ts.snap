// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/user should handle CREATE_USER 1`] = `
Object {
  "errors": "",
  "id": "",
  "items": Object {},
}
`;

exports[`reducers/user should handle DELETE_USER 1`] = `
Object {
  "errors": "",
  "id": "",
  "items": Object {},
}
`;

exports[`reducers/user should handle UPDATE_USER 1`] = `
Object {
  "errors": "",
  "id": "",
  "items": Object {},
}
`;

exports[`reducers/user should return the initial state 1`] = `
Object {
  "errors": "",
  "id": "",
  "items": Object {},
}
`;
